src/passes/Asyncify.cpp

```diff
@@ -163,6 +170,10 @@
 //    calls, so that you know when to start an asynchronous operation and
 //    when to propagate results back.
 //
+//  * asyncify_get_catch_counter(): call this to get the current value of the
+//    internal "__asyncify_catch_counter" variable (only when assertions
+//    or ignore mode are enabled).
```

Comment:
Just when `unreachable execution` happens you can call this function to see how many functions in stack uses catches before unwind. Like:
```
0 - unreachable happened because of some other assertion (probably not-in asyncify list)
1 - only one function involved
2 - two functions
3 - etc.
```

Not very handy because you can't understand the exact function names. But check for zero is useful, because there is no other way to understand if problem related to catches or non-asyncify function.

Better solution is to write in to console the reason of unreachable just before it happens, but idk is it possible somehow.

Reply:
Wouldn't the stack trace in the VM show the function names? I think I understand the extra info you are trying to add here, but if a user runs in a browser debugger I think they can get enough to debug with? The stack trace + they can see, visually, if they are inside a catch or not.